src/Runtime/OMResize.inc

```diff
@@ -0,0 +1,353 @@
+#ifdef __cplusplus
+#include <cassert>
+#else
+#include <assert.h>
+#endif
+
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "onnx-mlir/Runtime/OMTensor.h"
+
+/**
+ * The runtime implementation of ONNXResizeOp is manually translated python
+ * implementation: onnx/onnx/backend/test/case/node/resize.py (onnx v1.9.0)
+ * This implementation can be improved in efficiency.
+ **/
+
+static void linear_coeffs(float ratio, float *coeffs_buffer, int mode) {
+  coeffs_buffer[0] = 1 - ratio;
+  coeffs_buffer[1] = ratio;
+}
+
+static void nearest_coeffs(float ratio, float *coeffs_buffer, int mode) {
+  /* integer ratio is handled outside */
+  switch (mode) {
+  case 0: // round_prefer_float
+    coeffs_buffer[0] = ratio <= 0.5;
+    coeffs_buffer[1] = ratio > 0.5;
+    break;
+  case 1: // round_prefer_ceil
+    coeffs_buffer[0] = ratio < 0.5;
+    coeffs_buffer[1] = ratio >= 0.5;
+    break;
+  case 2: // floor
+    coeffs_buffer[0] = 1;
+    coeffs_buffer[1] = 0;
+    break;
+  case 4: // ceil
+    coeffs_buffer[0] = 0;
+    coeffs_buffer[1] = 1;
+    break;
+  }
+}
+
+static void cubic_coeffs(float ratio, float *coeffs_buffer, int mode) {
+  float A = -0.75;
+  // A may have different value for different coordinate_transformation_mode
+  // Currently, only default mode is supported
+  coeffs_buffer[0] =
+      ((A * (ratio + 1) - 5 * A) * (ratio + 1) + 8 * A) * (ratio + 1) - 4 * A;
+  coeffs_buffer[1] = ((A + 2) * ratio - (A + 3)) * ratio * ratio + 1;
+  coeffs_buffer[2] =
+      ((A + 2) * (1 - ratio) - (A + 3)) * (1 - ratio) * (1 - ratio) + 1;
+  coeffs_buffer[3] =
+      ((A * ((1 - ratio) + 1) - 5 * A) * ((1 - ratio) + 1) + 8 * A) *
+          ((1 - ratio) + 1) -
+      4 * A;
+}
+
+static void get_neighbor(float x, int64_t n, int limit, float *data,
```

Comment:
All the static functions in this file should have a comment describing what they do.